### DISCUSSION TITLE
Agilent 33500 - Source 2 control

### DISCUSSION THREAD

**Discussion by jcm-mx** [pymeasure/pymeasure#725] [Q&A] [answered]
Hi,

Awesome project, I'm using it with my agilent 33500b and it's making wonders. I'm curious if there is a way to control source 2. It seems like all the commands in the source code don't have the SOUR command to indicate what source gets data like amplitude, frequency, phase, etc.

If there is no way to do it, I'd be happy to work on it, as I said I have an agilent 33500b to test with.

Cheers

**Comment by msmttchr** [comment, marked as answer]
Hello, I think it's always very welcome to have people contributing to the project!
For the specific enhancement, I would first start from a general background on how instruments can be added (https://pymeasure.readthedocs.io/en/latest/dev/adding_instruments.html). It looks like to me that your extension could be handled with the new channel feature being discussed in #718 which I believe is not too far from completion. If you are not in a big hurry, you could plan your changes and then propose them when #718 is merged.
